### PR TITLE
Dfpl 671: Additional Question on Upload Additional Applications

### PR DESCRIPTION
**Pull request by SomeshRA**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DFPL-671

### Change description ###
Additional Question on Upload Additional Applications

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

**Review by chak-shing-lo-justice — APPROVED** [review]
LGTM

File: service/src/main/java/uk/gov/hmcts/reform/fpl/enums/UrgencyTImeFrameType.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.hmcts.reform.fpl.enums;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+public enum UrgencyTImeFrameType {
```
[the remark this answers is not in the document]
Comment (SomeshRA) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/enums/UrgencyTImeFrameType.java]: Corrected

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/common/OtherApplicationsBundle.java [hunk 2]
```diff
@@ -32,6 +33,7 @@
 public class OtherApplicationsBundle implements ApplicationsBundle {
     private final UUID id;
     private final OtherApplicationType applicationType;
+    private final UrgencyTImeFrameType urgencyTImeFrameType;
```
[the remark this answers is not in the document]
Comment (SomeshRA) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/common/OtherApplicationsBundle.java]: Corrected

**Review by so99y — APPROVED** [review]
Lgtm

**Review by so99y — APPROVED** [review]

**Review by so99y — DISMISSED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java [hunk 3]
```diff
@@ -117,12 +119,19 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
...
-        List<Element<DraftOrder>> draftOrders = caseData.getTemporaryC2Document().getDraftOrdersBundle();
-        unsealedCMOs.addAll(draftOrders.stream()
-            .map(Element::getValue)
-            .map(HearingOrder::from)
-            .map(ElementUtils::element)
-            .collect(Collectors.toList()));
+
+        Optional.ofNullable(caseData.getTemporaryC2Document())
+            .map(C2DocumentBundle::getDraftOrdersBundle)
+            .filter(not(Objects::isNull))
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:125]: Can it be simplified as `Objects.nonNull`?

Comment (SomeshRA) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java]: done!

**Review by so99y — APPROVED** [review]
LGTM

**Review by so99y — APPROVED** [review]

**Pull request merged**